java/code/src/com/suse/manager/reactor/PGEventStream.java

```diff
@@ -132,40 +158,49 @@ public void run() {
 
     @Override
     public void notification(int processId, String channelName, String payload) {
-        notification(Long.parseLong(payload));
+        List<Long> counts = Arrays.stream(payload.split(","))
+                .map(str -> Long.valueOf(str))
+                .collect(Collectors.toList());
+        notification(counts);
     }
 
     /**
      * Called every time a notification from Postgres (ultimately from mgr_engine.py) is fired.
-     * @param count the number of INSERTed events to be handled
+     * @param counts the list of the number of INSERTed events to be handled by queue. The first queue
+     *      is the one for events that aren't associated to a minion, queue -1.
      */
-    public void notification(long count) {
+    public void notification(List<Long> counts) {
         // compute the number of jobs we need to do - each job COMMITs individually
         // jobs = events / MAX_EVENTS_PER_COMMIT (rounded up)
-        long jobs = (count + MAX_EVENTS_PER_COMMIT - 1) / MAX_EVENTS_PER_COMMIT;
-
-        // queue one handlingTransaction(processEvents) call per job
-        for (int j = 0; j < jobs; j++) {
-            executorService.execute(() -> {
-                List<SaltEvent> uncommittedEvents = new LinkedList<>();
-                TransactionHelper.handlingTransaction(
-                        () -> processEvents(uncommittedEvents),
-                        e -> handleExceptions(uncommittedEvents, e));
-            });
+        for (final int queue : IntStream.range(-1, THREAD_POOL_SIZE).toArray()) {
+            long jobs = (counts.get(queue + 1) + MAX_EVENTS_PER_COMMIT - 1) / MAX_EVENTS_PER_COMMIT;
+
+            // queue one handlingTransaction(processEvents) call per job
+            for (int j = 0; j < jobs; j++) {
+                LOG.trace("Scheduling a job for queue " + queue);
+                ThreadPoolExecutor executor = queue < 0 ? executorGlobalService : executorServices.get(queue);
+                executor.execute(() -> {
+                    List<SaltEvent> uncommittedEvents = new LinkedList<>();
+                    TransactionHelper.handlingTransaction(
+                            () -> processEvents(uncommittedEvents, queue),
+                            e -> handleExceptions(uncommittedEvents, e));
+                });
+            }
         }
     }
 
     /**
      * Reads one or more events from suseSaltEvent and notifies listeners (typically, {@link PGEventListener}).
      * @param uncommittedEvents used to keep track of events being processed
+     * @param thread the index of the thread processing the events
```

Comment:
This should probably be `queue` meanwhile?

Reply:
Fixed